### PR TITLE
Introduce ::static_max_size() and max_size_v<T>

### PR DESCRIPTION
**Pull request by alexkaratarakis**

### CODE REVIEW

File: include/fixed_containers/enum_array.hpp [hunk 1]
```diff
@@ -172,3 +172,14 @@ class EnumArray
     constexpr ValueArrayType& values() { return IMPLEMENTATION_DETAIL_DO_NOT_USE_values_; }
 };
 }  // namespace fixed_containers
+
+// Specializations
+namespace std
+{
+template <typename L, typename T>
+struct tuple_size<fixed_containers::EnumArray<L, T>> : std::integral_constant<std::size_t, 0>
```
Comment (younesr1) [inline include/fixed_containers/enum_array.hpp:180]: So this prevents people from doing `auto [foo,bar,blah] = EnumArray<T,N>` and expecting it to work?

File: test/fixed_stack_test.cpp [hunk 2]
```diff
@@ -45,6 +45,13 @@ TEST(FixedStack, MaxSize)
         FixedStack<int, 3> v1{};
         EXPECT_EQ(3, v1.max_size());
     }
+
+    {
+        static_assert(FixedStack<int, 3>::static_max_size() == 3);
+        EXPECT_EQ(3, (FixedStack<int, 3>::static_max_size()));
```
Comment (younesr1) [inline test/fixed_stack_test.cpp:51]: why the runtime check right after the same compile time check? is this a convention?

**Review by younesr1 — APPROVED** [review]

File: include/fixed_containers/enum_array.hpp (hunk 1 above)
Comment (alexkaratarakis) [reply include/fixed_containers/enum_array.hpp:180]: Yeah, this will cause a compilation failure instead.

File: test/fixed_stack_test.cpp (hunk 2 above)
Comment (alexkaratarakis) [reply test/fixed_stack_test.cpp:51]: Coverage testing tools are runtime based and don't work with constexpr so I have both versions for coverage metrics.

File: include/fixed_containers/max_size.hpp [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <concepts>
+#include <cstddef>
+
+namespace fixed_containers
+{
+// Useful in generic code as std and other containers do not have `static_max_size()`
+// For std containers, this would be specialized to yield
+// `std::numeric_limits<difference_type>::max()`.
```
Comment (Bobobalink) [inline include/fixed_containers/max_size.hpp:10]: This intentionally doesn't include said specializations for `std::` types?
I was going to say we should add a note that the behavior you describe isn't *technically* correct, but I guess we leave that to the user.

Comment (alexkaratarakis) [reply include/fixed_containers/max_size.hpp:10]: Yeah, avoiding to drag-in `<vector>` and all other std containers just for this.

> I was going to say we should add a note that the behavior you describe isn't technically correct, but I guess we leave that to the user.

Isn't it? https://en.cppreference.com/w/cpp/container/vector/max_size

Comment (Bobobalink) [reply include/fixed_containers/max_size.hpp:10]: I was being pedantic, technically `std::` can do fancy things that make the limit smaller than we can do statically with `numeric_limits`

**Review by Bobobalink — APPROVED** [review]

**Pull request merged**